app/test/privacy_policy/table_of_contents/toc_section_expansion_test.dart

```diff
@@ -0,0 +1,593 @@
+              ],
+            ),
+          ];
+          tocController.build(sections);
+
+          // We expand it manually
+          tocController.toggleExpansionOfSection('Foo');
+          // We collapse it manually
+          tocController.toggleExpansionOfSection('Foo');
+          // "Scroll" to Foo
+          tocController.markAsCurrentlyRead('Foo');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+            ],
+          );
+        });
+      });
+      test('Collapsing and Expanding multiple sections works', () {
+        forExpansionBehavior(
+            [ExpansionBehavior.leaveManuallyOpenedSectionsOpen], () {
+          final sections = [
+            _Section(
+              'Foo',
+              subsections: const [
+                _Section('Bar'),
+                _Section('Baz'),
+              ],
+            ),
+            _Section(
+              'Quz',
+              subsections: const [
+                _Section('Xyzzy'),
+              ],
+            ),
+            _Section(
+              'Moa',
+              subsections: const [
+                _Section('Zuz'),
+              ],
+            )
+          ];
+          tocController.build(sections);
+
+          tocController.toggleExpansionOfSection('Foo');
+          tocController.toggleExpansionOfSection('Quz');
+          tocController.markAsCurrentlyRead('Moa');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+              _SectionResult('Quz', isExpanded: true),
+              _SectionResult('Moa', isExpanded: true),
+            ],
+          );
+        });
+      });
+
+      test('Regression test: Changing the expansion behavior works', () {
+        final sections = [
+          _Section(
+            'Foo',
+            subsections: const [
+              _Section('Bar'),
+              _Section('Baz'),
+            ],
+          ),
+        ];
+        tocController.changeExpansionBehaviorTo(
+            ExpansionBehavior.alwaysAutomaticallyCloseSectionsAgain);
+        tocController.build(sections);
+
+        tocController.toggleExpansionOfSection('Foo');
+        tocController.markAsCurrentlyRead('Foo');
+        tocController.markAsCurrentlyRead(null);
+
+        expect(
+          tocController.currentState.sections,
+          [
+            _SectionResult('Foo', isExpanded: false),
+          ],
+        );
+
+        tocController.changeExpansionBehaviorTo(
+            ExpansionBehavior.leaveManuallyOpenedSectionsOpen);
+
+        tocController.toggleExpansionOfSection('Foo');
+        tocController.markAsCurrentlyRead('Foo');
+        tocController.markAsCurrentlyRead(null);
+
+        expect(
+          tocController.currentState.sections,
+          [
+            _SectionResult('Foo', isExpanded: true),
+          ],
+        );
+      });
```

Comment:
```suggestion
      });

```

Add new line after test.